### ISSUE TITLE
This script is not newb-friendly

### ISSUE THREAD

**Issue by chadlinux** [LukeSmithxyz/emailwiz#319]
I have been following Luke Smith's content for the last few years. Love how much he's opened my eyes to the capabilities of free, open-source software.

That being said, the methods he suggests are not 100% newb-friendly. For example, this readme file says how to download the script, but he doesn't tell you to type `sh.emailwiz.sh` to get it running. It would be helpful if Luke would take extra care to show how to use this script out of the box all the way to using an email client such as thunderbird to access and read emails.

Are Luke's scripts and tutorials designed for computer-science students or professional programmers or are they supposed to be for the general public to try out? I have emailed him several times over the last few years. He never responds. I understand he puts these scripts out for free, but if he wants to offer solutions, it would be really helpful if he responded to questions via email or, better yet, github more frequently.

Even in [this video](https://videos.lukesmith.xyz/w/svCd4Y8Hghr4ecikLeGNQN), he says at the beginning, "I had better show this one particular detail or else I'll get 5000 emails". He knows that showing each step is important, but I've seen him make little mistakes along the way, which his audience sometimes corrects on his youtube channel.

Guess I'll have to use wordpress or something else off the shelf to make an email server. Disappointing that I can't get this stuff to work right :/

He wrote in the emailwiz readme file that you can test the DMARC settings etc using a third-party website, but he doesn't specify how. Even this command from [his related ](https://landchad.net/mail/validate/) article doesn't work when I ran his script with a fresh Vultr VPS server because my bash terminal said the "mail" command does not exist :/

```
echo "Hi there.

This is the text." | mail -s "Email from the server" your@emailaddress.com
```

Luke, I would be willing to donate a few bucks to you via zelle as thanks, but you said on your website that you will no longer accept "fiat money", only crypto such as bitcoin and monero. I don't do crypto. Not everyone does. I want to pay or donate to Luke to keep him motivated to answer support questions when his audience has them, but he doesn't seem very motivated to try that.

Guess I'll have to search for other github repos that have more eyeballs on them to receive support. Oh well. Still glad Luke published this script for those who have a better grip on how to use Linux terminal commands.

**Comment by TheGithubJoshua** [comment]
I would say running the script after installing it is obvious and doesn't need to be included.
And if `mail` says `command not found` this would indicate that the `mail` program is not installed on the system, installation instructions vary depending on the Linux system.
P.S you should probably look into setting up a Crypto wallet as it frees you from centralization and surveillance.

**Comment by chadlinux** [comment]
@TheGithubJoshua,

Thank you for your reply. I just signed up for github the other day. Did not know I could donate dirty fiat dollars to Luke as thanks for his efforts. I will definitely do that! Hard for him to stay motivated if there's no financial incentive.

I am trying to follow Luke's instructions to the letter wherever I can. I am a lifelong Macintosh user who's dabbled with command line on OSX and GNU/Linux, but no expert, as you can tell. I run Linux Mint 21.2 cinnamon on my desktop as my main OS, but strongly rely on the GUI to get things done.

I desperately need hand-holding to get my personal website up and running. I made more progress following Luke's landchat.net step-by-step "how to set up an email server" tutorial, than I have with the emailwiz script thus far. By following the landchat.net tutorials, I was able to shoot out an email from the root account of my Vultr VPS. 

1. I have a Vultr VPS (signed up with Luke's referral link). It's the cheapest option possible and runs  Debian 11 x64 (bullseye)
vCPU/s: 1 vCPU
RAM: 512.00 MB
Storage: 10 GB SSD  
2. I tried running emailwiz straight out of the box after running `apt update` and 'apt upgrade'. Luke didn't specify needing anything else installed, so I figured this would work straight out of the box. Should I have anything else configured on my server before running the script?
3. I am using Epik domains, which Luke recommended because they are pro-free speech. This is important for me too. I have my A/AAAA DNS records with Epik set the way he showed on landchad.net for the intro tutorial as you can see below.
![benpearce_xyz Epik domains A and AAAA records](https://github.com/LukeSmithxyz/emailwiz/assets/168049833/873ddc26-1485-4987-8f33-b1141eb5d572)
4. Are the landchad.net tutorials for setting up an email server supposed to match the emailwiz script? It seems like most of the steps overlap, broadly speaking.
5. When I create an email user such as 'billy', does that mean billy will have an email account on my personal server called billy@benpearce.xyz?
6. Can you explain how an email client such as thunderbird connects to a personal email server? Wish he provided screenshots, like he has on landchad.net because I got completely lost here using the landchad.net tutorials or the emailwiz script. 

Appreciate you answering my questions on Luke's behalf!

**Comment by TheGithubJoshua** [comment]
@chadlinux 

Email clients connect to an email client via SMTP (Simple Mail Transfer Protocol) which is a protocol like http or ftp
See https://en.wikipedia.org/wiki/Simple_Mail_Transfer_Protocol
As for No. 5 I am not sure but I would assume so.

**Comment by PhoenixBlade24** [comment]
I think Luke needs to make a genuine video tutorial on how to set up an email server with his script. All of his videos on the subject are either ironic speedrun style ones, or ones mixed in with web design. I just want a straightforward tutorial, from start-to-finish, on how to setup an email server in 2024 with emailwiz.

**Comment by chadlinux** [comment]
> I think Luke needs to make a genuine video tutorial on how to set up an email server with his script. All of his videos on the subject are either ironic speedrun style ones, or ones mixed in with web design. I just want a straightforward tutorial, from start-to-finish, on how to setup an email server in 2024 with emailwiz.

Yes! Completely agree. I am willing to donate money to Luke so he can refine the script or offer more tutorials to make it newb-friendly and 100% reliable.

I am tired of his ironic videos. He's made mistakes in those before. For example to do an SSH login without typing a password, he simply said "I already did this, just install pretty good privacy" which was unrelated to what he was trying to say! Of course, someone commented on his Youtube video that he made a mistake. Luke's response was "Ah shucks, thanks for pointing that out" :/

If his intended audience are linux geeks or IT professionals, that's fine. But when he claims to make things newb-friendly, try going the extra mile. You will be rewarded!

In the meantime, I bit the bullet and paid Epik domains for a Cpanel-controlled website that offers Let's Encrypt certificates and spam assassin for just $86/year. Beats the hassle, especially since this was for my business website. If he ever makes better tutorials, I can try again on a personal server.

He really needs to add step-by-step stuff. I'd donate $10/month to him, hoping he will, but I feel like he simply doesn't care. Too bad. Maybe he has something more profitable to do. If so, that's good.

**Comment by MatthewABrantley** [comment]
https://www.youtube.com/watch?v=3dIVesHEAzc

It's all there, step by step. If that is insufficient, I would strongly advise paying for professional in-person mentoring.

**Comment by chadlinux** [comment]
> https://www.youtube.com/watch?v=3dIVesHEAzc
> 
> It's all there, step by step. If that is insufficient, I would strongly advise paying for professional in-person mentoring.

I will look at this video. 

Where can one find professional, in-person Linux mentoring? I'd be happy to pay a Linux nerd for help to better understand the tricks and tips Luke recommends.

**Comment by P-B-H** [comment]
> > https://www.youtube.com/watch?v=3dIVesHEAzc
> > It's all there, step by step. If that is insufficient, I would strongly advise paying for professional in-person mentoring.
> 
> I will look at this video. This is much more in-depth than his jocular "speed run" emailwiz video.
> 
> Where can one find professional, in-person Linux mentoring? I'd be happy to pay a Linux nerd for help to better understand the tricks and tips Luke recommends.

Hi @chadlinux and @PhoenixBlade24! If you're looking for paid help, I have a web hosting company (parkbenchhosting.com) where we specialize in setting up very small businesses with no prior web presence at all and usually little to no knowledge of how to create one. I'm sure we could get you set up with a simple website and email that you could be confident and proud of.

**Comment by LukeSmithxyz** [comment]
This definitely shouldn't be an entirely "newb-friendly" script.  The target audience is recreational power-users, like myself. I mean if a person doesn't know how to download a file from a URL or run a shell script, they really might want to wait before running an email server.

If I get the time and will to dive into this script again in detail, perhaps I will (although it is constantly growing in features). But like LARBS, ultimately this script came from my own personal desire to get simple things done quick for myself. Others in a similar position can use it.

Either way, my YouTube has never been a consultancy firm. It's only even peripherally about technology, even though I've done a lot of that. I go over what I think is fun for people, or might open doors, but they must do the work. Other people do more hand-holding things and that's cool, but I got a lot of other things right now. I'd like to scratch this itch for you and explain every command, but I can't right now. Maybe sometime I will, but I couldn't even record an hours long stream explaining all this right now if I could.
